collectors/googlestackdriver/collector.js

```diff
@@ -86,7 +88,22 @@ timestamp < "${state.until}"`;
         const paginationCallback = (result, acc = []) => {
             AlLogger.info(`Getting page: ${pagesRetireved + 1} Logs retrieved: ${result[0].length}`);
             pagesRetireved++;
-            const logs = result[0];
+            let logs = result[0].map(logEntry => {
+                // Decode protoPayload buffer if it's a LogEntry google.cloud.audit.AuditLog
+                if (logEntry.protoPayload && (logEntry.protoPayload.type_url === 'type.googleapis.com/google.cloud.audit.AuditLog')) {
+                    const protoPath = `./node_modules/google-proto-files/google/cloud/audit/audit_log.proto`;
+                    const root = protoFiles.loadSync(protoPath);
+                    const AuditLog = root.lookupType('google.cloud.audit.AuditLog');
+                    try {
+                        const buffer = Buffer.from(logEntry.protoPayload.value);
+                        let decodedData = AuditLog.decode(buffer);
+                        logEntry.protoPayload.value = decodedData.toJSON();
+                    } catch(error) {
+                        AlLogger.error(`Error decoding data ${error}`);
+                    }
+                }
+                return logEntry;
+            });
```

Comment:
could we please move the decoding logic to a separate function/method? I would also suggest that we only load the `AuditLog` proto stuff once and then re-use it -- I think the current implementation would load it for every audit log msg we process

Reply:
made changes